### ISSUE TITLE
More glue fails and sneak case

### ISSUE THREAD

**Issue by latot** [r-lib/lintr#1466]
Hi, I was running linting and found this:

test.R
```
df <- data.frame(list(
a = c(1, 2, 3)
))

v <- "a"

dplyr::mutate(df, !!paste0(tolower(v), "_l") := df$a * 2)
```

The output of the linting:
```
test.r:7:31: style: [object_name_linter] Variable and function name style should be snake_case or symbols.
dplyr::mutate(df, !!paste0(v, "_l") := df$a * 2)
```

Thx!

**Comment by IndrajeetPatil** [comment]
Reprex:

``` r
library(lintr)

df <- dplyr::tibble(a = c(1))
v <- "a"

dplyr::mutate(df, !!paste0(tolower(v), "_l") := df$a * 2)
#> # A tibble: 1 × 2
#>       a   a_l
#>   <dbl> <dbl>
#> 1     1     2

lint(
  text = 'dplyr::mutate(df, !!paste0(v, "_l") := df$a * 2)',
  linter = object_name_linter()
)
#> <text>:1:31: style: [object_name_linter] Variable and function name style should be snake_case or symbols.
#> dplyr::mutate(df, !!paste0(v, "_l") := df$a * 2)
#>                               ^~~~
```

<sup>Created on 2022-07-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
